### PR TITLE
Code, Unit tests, and Javadoc for Assignment one 

### PR DESCRIPTION
**Pull request by MikeBrooksNH**
Not sure if you needed to also be designated as the reviewer

not sure about the way the directory layout should actually be...

I didn't include a test for every method...

Javadoc is pretty basic...

homework 1 done - forgot to assign the pull request...

### CODE REVIEW

File: src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+package edu.mbrooks.advancedjava.impl;
+
+import edu.mbrooks.advancedjava.interfaces.Icollectable;
+
+/**
+ *
+ * @author michaelbrooks
+ * This is a class to setup if there is a collection of valuables, or a bunch of trash
+ */
+public class CollectableImpl implements Icollectable {
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java]: as you find out as you take this course, I am pretty picky about the names of things. one of things I do not like (even though it is somewhat common) is to use the suffix "impl" on class names. Impl is a detail that should be obvious from the context of the classes' use. It makes the class name less English like and therefore harder to read. 
Basically, you don't need it. You can determine if the class is an interface or a concrete class without this suffix.

File: src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    String pileOf;
+    boolean getRich;
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java]: this should be private 
also Boolean variables are typically hasXXX or isXXX not sure what getRich (true or false) means.

File: src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        this.getRich = ofValue;
+    }
+
+    @Override
+    /**
+     * putting the collection type and realative value posibility into a string
+     */
+    public String toString() {
+
+        String retVal = this.pileOf;
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java]: use whole word variables makes code much easier to read.

File: src/main/java/edu/mbrooks/advancedjava/impl/Person.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
+/**
+ * Program Name:    Person.java
+ * Author:          Michael Brooks
+ * Version:         Version 1.0
+ * Course:          Advanced Java Programming, Summer 2018
+ * Copyright:       (C) 2018 Michael A. Brooks
+ * Description:     Implementation of the person interface
+ */
+package edu.mbrooks.advancedjava.impl;
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/impl/Person.java]: impl definitely should not be a package name. Interfaces and Implementations should typically exist in the same package.

File: src/main/java/edu/mbrooks/advancedjava/impl/Person.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
+/**
+ * Program Name:    Person.java
+ * Author:          Michael Brooks
+ * Version:         Version 1.0
+ * Course:          Advanced Java Programming, Summer 2018
+ * Copyright:       (C) 2018 Michael A. Brooks
+ * Description:     Implementation of the person interface
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/impl/Person.java]: It is ok to have a file header with meta data like copyright but the description should go below in the class JavaDoc documentation.  Also, the file header doc should be not be JavaDoc

File: src/main/java/edu/mbrooks/advancedjava/impl/Person.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+ */
+public class Person implements Iperson {
+
+    private String name;
+    private String address;
+    private String ssn;
+    private java.util.GregorianCalendar dob;
+
+    /**
+     * Default constructor sets the name to an empty string
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/impl/Person.java]: technically this is not a default constructor. The default constructor takes no arguments. If a class has no constructors at all, the default constructor is automatically generated.

File: src/main/java/edu/mbrooks/advancedjava/impl/Person.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    /**
+     *
+     * @return String formatted as a 9 digit string in the form of 000-00-0000
+     */
+
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/impl/Person.java]: careful with spacing.

File: src/main/java/edu/mbrooks/advancedjava/interfaces/Icollectable.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package edu.mbrooks.advancedjava.interfaces;
+
+/**
+ *
+ * @author michaelbrooks
+ * This is a class to setup the basics of what a general collection informaiton is should have
+ */
+public interface Icollectable {
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/interfaces/Icollectable.java]: Don't prefix interfaces with I. Interface or class should be and typically is easily conferred by context. Also make the code harder to read.

File: src/main/java/edu/mbrooks/advancedjava/interfaces/Iperson.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+    // getter / setter for the person's name
+    String getName();
+    void setName(String name);
+
+    // getter / setter for the address of the person
+    String getAddress();
+    void setAddress(String address);
+
+    // getter / setter for the person's SSN
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/interfaces/Iperson.java]: use java doc here like you did elsewhere

File: src/main/java/edu/mbrooks/advancedjava/main/Collectable.java [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+ * Course:          Advanced Java Programming, Summer 2018
+ * Copyright:       (C) 2018 Michael A. Brooks
+ * Description:     Implementation of the collectable interface
+ */
+package edu.mbrooks.advancedjava.main;
+
+/**
+ *
+ * @author michaelbrooks
+ * This is a collection of something
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/main/Collectable.java]: this is too vague.
why is this class needed? how does it differ from all the other collection classes.

File: src/main/java/edu/mbrooks/advancedjava/main/Collectable.java [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+ */
+public class Collectable {
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/main/Collectable.java]: Name should be specific

File: src/main/java/edu/mbrooks/advancedjava/main/Collectable.java [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+        return faceValue;
+    }
+
+    /**
+     *
+     * @param String face faceValue of the item (i.e. the Penny Black is 1 British Cent)
+     */
+    public void setFaceValue(String faceValue) {
+        this.faceValue = faceValue;
+    }
```
Comment (smarks) [inline src/main/java/edu/mbrooks/advancedjava/main/Collectable.java]: generally a good idea to provide a toString method.

**Comment by smarks** [comment]
Nice work here. !

File: src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java (hunk 1 above)
Comment (MikeBrooksNH) [reply src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java]: Understand.

File: src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java (hunk 2 above)
Comment (MikeBrooksNH) [reply src/main/java/edu/mbrooks/advancedjava/impl/CollectableImpl.java]: Never saw that before, I like that, makes perfect sense

File: src/main/java/edu/mbrooks/advancedjava/impl/Person.java (hunk 4 above)
Comment (MikeBrooksNH) [reply src/main/java/edu/mbrooks/advancedjava/impl/Person.java]: This is the coding standard I've seen where I work... I'll keep this in mind for the future - also answers the question I posted with the assignment

File: src/main/java/edu/mbrooks/advancedjava/main/Collectable.java (hunk 11 above)
Comment (MikeBrooksNH) [reply src/main/java/edu/mbrooks/advancedjava/main/Collectable.java]: Comment is bad...  This is a single collectable item - the base class for a collection of things, or a collectables - the derived class will be a collection of stamps, coins, porcelain figures, fishing poles, records, etc. 

I think it meets the "is a" rule:
          A Stamp "is a" Collectable.
          A Coin "is a" Collectable.
          A Baseball Card "is a" Collectable.

Do I have that wrong?

File: src/main/java/edu/mbrooks/advancedjava/main/Collectable.java (hunk 12 above)
Comment (MikeBrooksNH) [reply src/main/java/edu/mbrooks/advancedjava/main/Collectable.java]: Shoot! that was an easy one! Can't believe I missed that!!! Too focused on the derived class - Stamp has a .toString() method, but that should've had a call to the base class.

**Comment by MikeBrooksNH** [comment]
Thanks for the great feedback!

**Pull request merged**